data/dataset.py

```diff
@@ -0,0 +1,39 @@
+import json
+from abc import abstractmethod
+
+from datasets import Dataset as HuggingFaceDataset
+from torch.utils.data import Dataset as TorchDataset
+from object_generators.tokenizer_generator import TokenizerGenerator
+
+
+class MidiDataset(TorchDataset):
+    def __init__(
+        self,
+        dataset: HuggingFaceDataset,
+    ):
+        super().__init__()
+
+        # Dataset with tokenized MIDI data
+        self.dataset = dataset
+        tokenizer_info = json.loads(dataset.description)
+        tokenizer_name = tokenizer_info["tokenizer_name"]
+        tokenizer_parameters = tokenizer_info["tokenizer_parameters"]
+
+        tokenzier_generator = TokenizerGenerator()
+
+        self.tokenizer = tokenzier_generator.generate_tokenizer(
+            tokenizer_name,
+            tokenizer_parameters,
+        )
```

Comment:
```suggestion
        self.tokenizer = TokenizerGenerator.generate_tokenizer(
            tokenizer_name,
            tokenizer_parameters,
        )
```

I'm still struggling to see the point of this class :)

Reply:
Allows us to use any tokenization logic in the dataset, makes MidiDataset class much more versatile and gives us more space to edit midi_tokenizers :).

Reply:
But what's the point of the class? Looks like a method `generate_tokenizer` would be able to do all the same things

Reply:
oh